lib/preset-patterns.js

```diff
@@ -18,10 +18,10 @@ module.exports = {
  */
 function suitSelector(componentName, presetOptions) {
   var ns = (presetOptions && presetOptions.namespace) ? presetOptions.namespace + '-' : '';
-  var OPTIONAL_PART = '(?:-[a-zA-Z0-9]+)?';
-  var OPTIONAL_MODIFIER = '(?:--[a-zA-Z0-9]+)?';
+  var OPTIONAL_PART = '(?:-[a-z][a-zA-Z0-9]*)?';
+  var OPTIONAL_MODIFIER = '(?:--[a-z0-9][a-zA-Z0-9]*)?';
   var OPTIONAL_ATTRIBUTE = '(?:\\[\\S+\\])?';
-  var OPTIONAL_STATE = '(?:\\.is-[a-zA-Z0-9]+)?';
+  var OPTIONAL_STATE = '(?:\\.is-[a-z][a-zA-Z0-9]*)?';
```

Comment:
They're not common, but I don't think they would be an anti-pattern. So perhaps we should let those through